tests/v2/validation/provisioning/rke2/hostname_truncation.go

```diff
@@ -0,0 +1,151 @@
+package rke2
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"testing"
+
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	steveV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	nodestat "github.com/rancher/rancher/tests/framework/extensions/nodes"
+	"github.com/rancher/rancher/tests/framework/extensions/pipeline"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/environmentflag"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	name2 "github.com/rancher/wrangler/pkg/name"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+func TestHostnameTruncation(t *testing.T, client *rancher.Client, provider Provider, machinePools []provv1.RKEMachinePool, defaultHostnameLengthLimit int, kubeVersion, cni string) {
+	cloudCredential, err := provider.CloudCredFunc(client)
+	require.NoError(t, err)
+
+	generatedPoolName := "nc-"
+	machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+	machineConfigResp, err := client.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+	require.NoError(t, err)
+
+	for i := range machinePools {
+		machinePools[i].NodeConfig = &corev1.ObjectReference{
+			Kind: machineConfigResp.Kind,
+			Name: machineConfigResp.Name,
+		}
+		machinePools[i].EtcdRole = true
+		machinePools[i].ControlPlaneRole = true
+		machinePools[i].WorkerRole = true
+	}
+
+	cluster := clusters.NewK3SRKE2ClusterConfig("", namespace, cni, cloudCredential.ID, kubeVersion, "", machinePools)
+
+	cluster.GenerateName = "t-"
+	cluster.Spec.RKEConfig.MachinePoolDefaults.HostnameLengthLimit = defaultHostnameLengthLimit
+	clusterResp, err := clusters.CreateK3SRKE2Cluster(client, cluster)
+	require.NoError(t, err)
+
+	require.NoError(t, steveV1.ConvertToK8sType(clusterResp.JSONResp, cluster))
+	clusterName := cluster.Name
+	assert.True(t, len(clusterName) == 7)
+
+	if client.Flags.GetValue(environmentflag.UpdateClusterName) {
```

Comment:
This flag isn't really necessary, we only use this if we're linking the provisioned cluster to another test. Since this is dealing with the hostname truncation, we probably don't need this.